ops/framework.py

```diff
@@ -836,6 +921,7 @@ def __exit__(self, exception_type, exception, traceback):
                 self._storage.drop_notice(event_path, observer_path, method_name)
                 continue
 
+            event = typing.cast(EventBase, event)
```

Comment:
It really seems like there should be a good way to genericify things somehow so we don't need these casts, but I couldn't come up with any ideas after staring/trying for a bit.  Oh well...

Reply:
yeah, same... let's keep thinking on it :D